code/authenticator/RESTfulAPI_TokenAuthenticator.php

```diff
@@ -344,6 +319,40 @@ private function generateToken($expired = false)
     ); 
   }
 
+  /**
+   * Update the token of a token owner
+   * @param $owner          The token owner instance to update
+   * @param bool $expired   Set to true to generate an outdated token
+   * @return array|null     Token data array('token' => HASH, 'expire' => EXPIRY_DATE)
+   */
+  private function updateToken($owner, $expired = false)
+  {
+    $ownerId = null;
+    $tokenData = null;
+
+    // DB field names
+    $tokenDBColumn  = $this->tokenConfig['DBColumn'];
+    $expireDBColumn = $this->tokenConfig['expireDBColumn'];
+
+    do {
+      try {
+        //generate token
+        $tokenData = $this->generateToken( $expired );
+
+        // ensure we never regenerate the same token!
+        if($owner->{$tokenDBColumn} != $tokenData['token']){
```

Comment:
The point where the token would collide is in the Database. So, if we add the type check, do we improve or worsen the stability of the check?

The only distinction on the DB side would be 'null' != null? Otherwise everything should be treated as string (varchar). I think it's sensible to assume both tokens will become strings, so I'd stick with the `!=` check, since checking for type inequality isn't important to the DB... or am I missing something important here?